sklearn/neighbors/tests/test_kde.py

```diff
@@ -141,3 +141,44 @@ def test_kde_pipeline_gridsearch():
     search = GridSearchCV(pipe1, param_grid=params, cv=5)
     search.fit(X)
     assert_equal(search.best_params_['kerneldensity__bandwidth'], .1)
+
+
+def test_kde_sample_weights():
+    N = 10000
+    T = 100
+    W_neutral = 3 * np.ones(N)
+    for d in [1, 2, 10]:
+        rng = np.random.RandomState(0)
+        X = rng.rand(N, d)
+        W = 1 + (10 * X.sum(axis=1)).astype(np.int8)
+        repetitions = []
+        for x, w in zip(X, W):
+            for _ in range(w):
+                repetitions.append(x.tolist())
+        X_repetitions = np.array(repetitions)
+        Y = rng.rand(T // d, d)
```

Comment:
Why `T // d` ?

Reply:
We want the first argument (the number of test points) to be an integer, both in python2 and python3.

Reply:
Yes but why would you divide T by d? It is just another integer that you could name `n_samples_2`, isn't it?

Reply:
Yes you're right, I'll rename this to make it more expressive.
I do this mainly because I want to keep the number of operations under control while increasing the dimension of the space, so I decrease the cardinality of the test sample.